modules/cli/src/main/scala/scala/cli/commands/InstallHome.scala

```diff
@@ -4,64 +4,96 @@ import caseapp._
 import coursier.env.{EnvironmentUpdate, ProfileUpdater}
 
 import scala.io.StdIn.readLine
-import scala.util.Properties
+import scala.util.{Properties, Try}
 
 object InstallHome extends ScalaCommand[InstallHomeOptions] {
   override def hidden: Boolean = true
+
+  private def isOutOfDate(newVersion: String, oldVersion: String): Boolean = {
+    val versionOrdering = Ordering.by { (_: String).split("""\.""").map(_.toInt).toIterable }
+    versionOrdering.gt(newVersion, oldVersion)
+  }
+
   def run(options: InstallHomeOptions, args: RemainingArgs): Unit = {
 
-    val binDirPath      = scala.build.Directories.default().binRepoDir
-    val scalaCliBinPath = binDirPath / "scala-cli"
+    val binDirPath =
+      options.binDirPath.getOrElse(scala.build.Directories.default().binRepoDir / "scala-cli")
+    val newScalaCliBinPath = os.Path(options.scalaCliBinaryPath, os.pwd)
+
+    val newVersion: String =
+      os.proc(newScalaCliBinPath, "about", "--version").call(cwd = os.pwd).out.text.trim
 
-    if (os.exists(scalaCliBinPath)) {
-      if (options.force) os.remove.all(scalaCliBinPath)
-      else if (coursier.paths.Util.useAnsiOutput()) {
-        println(
-          "scala-cli already exists. Do you want to override it [Y/n]"
+    // Backward compatibility - previous versions not have the `--version` parameter
+    val oldVersion: String = Try {
+      os.proc(binDirPath / options.binaryName, "about", "--version").call(cwd =
+        os.pwd).out.text.trim
+    }.toOption.getOrElse("0.0.0")
+
+    if (os.exists(binDirPath)) {
```

Comment:
Would it be possible to flatten these nested ifs? The logic here is quite sophisticated and it might be easy to get lost in futre